### PR TITLE
Add MacOS support

### PR DESCRIPTION
**Pull request by suntabu**

### CODE REVIEW

File: Assets/NuGet/Editor/NugetHelper.cs [hunk 1]
```diff
@@ -208,6 +211,21 @@ private static void RunNugetProcess(string arguments, bool logOuput = true)
 
             LogVerbose("Running: {0} \nArgs: {1}", files[0], arguments);
 
+#if UNITY_EDITOR_OSX
+            var monoFile = "/Library/Frameworks/Mono.framework/Versions/Current/Commands/mono";
+            //TODO:using unity mono to pack the Nuget package.
+            var commandLine = " " + files[0] + " " + arguments;
```
Comment (jwittner) [inline Assets/NuGet/Editor/NugetHelper.cs]: Seems like more of this code could be shared between the two platforms. Looks like only the `FileName` and `Arguments` are different?

File: Assets/NuGet/Editor/NugetHelper.cs [hunk 2]
```diff
@@ -36,8 +36,11 @@ public static class NugetHelper
         /// <summary>
         /// The path where to put created (packed) and downloaded (not installed yet) .nupkg files.
         /// </summary>
+#if UNITY_EDITOR_OSX
+        public static readonly string PackOutputDirectory = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData) + "/NuGet/Cache";
```
Comment (jwittner) [inline Assets/NuGet/Editor/NugetHelper.cs]: Would `Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),"NuGet","Cache")` work on all platforms?

File: Assets/NuGet/Editor/NugetHelper.cs [hunk 3]
```diff
@@ -208,6 +211,21 @@ private static void RunNugetProcess(string arguments, bool logOuput = true)
 
             LogVerbose("Running: {0} \nArgs: {1}", files[0], arguments);
 
+#if UNITY_EDITOR_OSX
+            var monoFile = "/Library/Frameworks/Mono.framework/Versions/Current/Commands/mono";
+            //TODO:using unity mono to pack the Nuget package.
```
Comment (jwittner) [inline Assets/NuGet/Editor/NugetHelper.cs]: Can you drop this TODO and leave a comment that we're using mono to run 'nuget.exe' - this method is used for all nuget interfacing - not just to pack things.

File: Assets/NuGet/Editor/NugetHelper.cs (hunk 1 above)
Comment (suntabu) [reply Assets/NuGet/Editor/NugetHelper.cs]: Done

File: Assets/NuGet/Editor/NugetHelper.cs (hunk 2 above)
Comment (suntabu) [reply Assets/NuGet/Editor/NugetHelper.cs]: Yes, but 2 Path.Combine invokes would be used

File: Assets/NuGet/Editor/NugetHelper.cs (hunk 3 above)
Comment (suntabu) [reply Assets/NuGet/Editor/NugetHelper.cs]: Done

File: Assets/NuGet/Editor/NugetHelper.cs (hunk 2 above)
Comment (jwittner) [reply Assets/NuGet/Editor/NugetHelper.cs]: That seems ok right?

File: Assets/NuGet/Editor/NugetHelper.cs (hunk 1 above)
Comment (jwittner) [reply Assets/NuGet/Editor/NugetHelper.cs]: Yay!

File: Assets/NuGet/Editor/NugetHelper.cs (hunk 2 above)
Comment (suntabu) [reply Assets/NuGet/Editor/NugetHelper.cs]: Yes, only 2 arguments needed

Comment (jwittner) [reply Assets/NuGet/Editor/NugetHelper.cs]: Thanks for the change! =)

**Pull request merged**